app/controllers/AccountManagement.scala

```diff
@@ -296,18 +298,18 @@ object ManageWeekly extends ContextLogging {
             returnError(e.getMessage, sub)
         }
       }
-      response.valueOr(error => Future(BadRequest(jsonError(error))))
+      response.run.map(_.valueOr(error => Future(BadRequest(jsonError(error))))).flatMap(identity)
```

Comment:
there's no flatten on a future and I ended up with 2 futures in different places in the stack